src/Downloader.php

```diff
@@ -24,46 +54,176 @@ public function __construct($host, $port, $username, $password)
         $this->port = $port;
         $this->username = $username;
         $this->password = $password;
+
+        if ($defaultProcessAction !== null and $defaultProcessAction instanceof ProcessAction) {
+            $this->defaultProcessAction = $defaultProcessAction;
+        } else {
+            $this->defaultProcessAction = ProcessAction::move('INBOX/processed');
+        }
+    }
+
+    /**
+     * Set inbox folder
+     * @param string $inboxFolder
+     * @default "INBOX"
+     * @return $this
+     */
+    public function setInboxFolder($inboxFolder = 'INBOX')
+    {
+        $this->inboxFolder = $inboxFolder;
+        return $this;
+    }
+
+    /**
+     * @param bool $enabled
+     * @return $this
+     */
+    public function setProcessedFoldersAutomake($enabled)
+    {
+        $this->processedFoldersAutomake = $enabled;
+        return $this;
+    }
+
+    /**
+     * @param ProcessAction $processAction
+     * @return $this
+     * @throws \Exception
+     */
+    public function setDefaultProcessAction(ProcessAction $processAction)
+    {
+        if ($processAction === null or !($processAction instanceof  ProcessAction)) {
+            throw new \Exception('Default processed action is invalid!');
+        }
+        $this->defaultProcessAction = $processAction;
+        return $this;
     }
 
-    public function fetch(MailCriteria $criteria, $callback)
+    /** Get IMAP alerts
+     * @return array|bool
+     * @see imap_alerts()
+     */
+    public function getAlerts()
     {
-        $mailbox = @imap_open('{' . $this->host . ':' . $this->port . '}INBOX', $this->username, $this->password);
-        if (!$mailbox) {
-            throw new ImapException("Cannot connect to imap server: {$this->host}:{$this->port}'");
+        return $this->alerts;
+    }
+
+    /** Get IMAP errors
+     * @return array|bool
+     * @see imap_errors()
+     */
+    public function getErrors()
+    {
+        return $this->errors;
+    }
+
+    public function fetch(MailCriteria $criteria, $callback, $fetchParts = null)
```

Comment:
This method starts looking like very long and complicated method. This should be refactored. It is not part of this merge request ;-) i am just writing it for future myself.